src/webportal/src/app/job-submission/models/protocol-schema.js

```diff
@@ -67,7 +67,7 @@ export const taskRoleSchema = Joi.object().keys({
 });
 
 export const taskRolesSchema = Joi.object().pattern(
-  /^[a-zA-Z_][a-zA-Z0-9_]*$/,
+  /^[a-z][a-z0-9]*$/,
```

Comment:
Please refer https://github.com/microsoft/frameworkcontroller/blob/d602cdcbab2b91a7772f9c0d956d17e5f9d69847/pkg/apis/frameworkcontroller/v1/crd.go#L33 for task role name